python/cuml/cuml/accel/tests/xfail_manager.py

```diff
@@ -0,0 +1,873 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Xfail list management tool for cuml.accel tests.
+
+This module provides tools for editing and maintaining xfail lists used in
+cuml.accel testing. It offers both a programmatic API through the XfailManager
+and XfailGroup classes, and a command-line interface for common operations.
+
+Key features:
+- Edit test properties and move tests between groups
+- Add/remove tests from existing groups
+- Deterministic formatting and sorting of xfail lists
+- Support for @file syntax to read test IDs from files
+- Validation of test IDs and group conditions
+
+CLI Commands:
+- edit-tests: Modify properties of tests and move them to new groups
+- add-tests: Add tests to existing groups
+- remove-tests: Remove tests from xfail lists
+- format: Apply consistent formatting and sorting
+
+The tool ensures xfail lists remain maintainable and produce clean diffs
+in version control systems.
+"""
+
+import argparse
+import re
+import sys
+from collections import OrderedDict, defaultdict
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import yaml
+from packaging.requirements import Requirement
+
+
+class QuoteTestID(str):
+    """String subclass to force quoting of test IDs."""
+
+    pass
+
+
+def setup_yaml():
+    """Configure YAML dumper with custom string handling."""
+
+    def quoted_scalar(dumper, data):
+        scalar_tag = "tag:yaml.org,2002:str"
+        return dumper.represent_scalar(scalar_tag, data, style='"')
+
+    yaml.add_representer(QuoteTestID, quoted_scalar)
+    yaml.add_representer(
+        OrderedDict,
+        lambda dumper, data: dumper.represent_mapping(
+            "tag:yaml.org,2002:map", data.items()
+        ),
+    )
+
+
+class XfailGroup:
+    """Represents a group of related test failures with common characteristics."""
+
+    def __init__(
+        self,
+        reason: str,
+        tests: Optional[List[str]] = None,
+        strict: bool = True,
+        condition: Optional[str] = None,
+        marker: Optional[str] = None,
```

Comment:
This is still missing the recently introduced `run:` property.

Reply:
Added in https://github.com/rapidsai/cuml/pull/6892/commits/27f730fd6d904a0a324ef8325be34640bacfc21e .